File: Java/Easy/The_Major_Element/the_major_element.java
import java.util.HashMap;

public class the_major_element {
	public static void main (String[] args) {
		String line = "92,19,19,76,19,21,19,85,19,19,19,94,19,19,22,67,83,19,19,54,59,1,19,19";
		System.out.println(funcName(line));
	}
	
	public static String funcName(String line) {
		String[] line_array = line.split(",");
		float length = line_array.length;
		HashMap<String, Float> hm = new HashMap<String, Float>();
		boolean flag = false;
		for (String num:line_array) {
			float count = 1;
			if (hm.containsKey(num)) {
				count = hm.get(num);
				count += 1;
			}
			hm.put(num, count);
			if ((count / length) >= 0.5) {
				return num;
			}
		}
		if (flag == false) {
			return "None";
		}
		return null;
	}
}

File: Python/Easy/the_major_element.py
line = "92,19,19,76,19,21,19,85,19,19,19,94,19,19,22,67,83,19,19,54,59,1,19,19"


def major_element(line):
    line_array = line.split(",")
    minimum_count = len(line_array) / 2
    my_dict = {}
    flag = False
    # pu.db
    for num in line_array:
        count = 1
        if num in my_dict:
            count = my_dict.get(num)
            count += 1
        my_dict[num] = count
        if count >= minimum_count:
            return num
    if flag is False:
        return "None"

print(major_element(line))
